src/Ship.cpp

```diff
@@ -361,6 +366,65 @@ void Ship::UpdateMass()
 	SetMass((m_stats.static_mass + GetPropulsion()->FuelTankMassLeft()) * 1000);
 }
 
+//calculates atmospheric lift
+vector3d Ship::CalcAtmoLift()
+{
+	double m_topCrossSec = GetShipType()->topCrossSection;
+	double m_shipLiftCoeff = GetShipType()->shipLiftCoefficient;
+
+	double m_fDrag = CalcAtmosphericForce(DEFAULT_DRAG_COEFF); //returns 0 if not in atmosphere, so no need to check
+
+	vector3d m_AoAVector = GetOrient().VectorZ().NormalizedSafe() - GetVelocity().NormalizedSafe();
```

Comment:
I may suggest to find the angle between these vectors?

`GetOrient().VectorZ()` don't needs `NormalizedSafe()`
Instead, in `GetVelocity` you may have a "0" vector,
but in that case I think you could "bail out" because your
ship is almost immbile, isn't?

Reply:
It doesn't have to be directly proportional to the angle, this method makes the derivative of the torque applied change with the angle between vectors, so I that's why I used it this way. Feels less artifical to me. Also, the AoA vector isn't directly used in anywhere right now, so I though I could keep it like that until I got my head into the actual complex stuff.

Thanks, I will remove the NormalizedSafe().

The AoA Vector itself isn't directly used anyway, so it doesn't really matter when the velocity vector approaches 0.